chat-android/src/main/java/com/ably/chat/Discontinuities.kt

```diff
@@ -0,0 +1,55 @@
+package com.ably.chat
+
+import io.ably.lib.types.ErrorInfo
+import io.ably.lib.util.EventEmitter
+import io.ably.lib.util.Log
+import io.ably.lib.realtime.ChannelBase as AblyRealtimeChannel
+
+/**
+ * Represents an object that has a channel and therefore may care about discontinuities.
+ */
+interface HandlesDiscontinuity {
+    /**
+     * A promise of the channel that this object is associated with. The promise
+     * is resolved when the feature has finished initializing.
+     */
+    val channel: AblyRealtimeChannel
+
+    /**
+     * Called when a discontinuity is detected on the channel.
+     * @param reason The error that caused the discontinuity.
+     */
+    fun discontinuityDetected(reason: ErrorInfo?)
+}
+
+/**
+ * An interface to be implemented by objects that can emit discontinuities to listeners.
+ */
+interface EmitsDiscontinuities {
+    /**
+     * Register a listener to be called when a discontinuity is detected.
+     * @param listener The listener to be called when a discontinuity is detected.
+     */
+    fun onDiscontinuity(listener: Listener): Subscription
+
+    /**
+     * An interface for listening when discontinuity happens
+     */
+    fun interface Listener {
+        /**
+         * A function that can be called when discontinuity happens.
+         * @param reason reason for discontinuity
+         */
+        fun discontinuityEmitted(reason: ErrorInfo?)
+    }
+}
+
+open class DiscontinuityEmitter : EventEmitter<String, EmitsDiscontinuities.Listener>() {
```

Comment:
I think it would be much better if the class were final. I can’t find any place where we inherit from it

Reply:
Fixed https://github.com/ably-labs/ably-chat-kotlin/pull/33/commits/0aac6edfb3570c08532e0615c85f428748773eac